state_synchronizer/src/coordinator.rs

```diff
@@ -336,4 +350,30 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         req.set_ledger_info_with_sigs(ledger_info.into_proto());
         self.executor_proxy.execute_chunk(req).await
     }
+
+    async fn check_subscriptions(&mut self) -> Result<()> {
+        let ledger_info = self.executor_proxy.get_latest_ledger_info().await?;
+        let committed_version = self.known_version;
+        let mut ready = HashMap::new();
+
+        self.subscriptions
+            .retain(|peer_id, (expiry, known_version, limit)| {
+                // filter out expired peer requests
+                if SystemTime::now().duration_since(expiry.clone()).is_ok() {
+                    return false;
+                }
+                if *known_version < committed_version {
```

Comment:
For clarity, it might be better to do the following: 
```rust
if *known_version < committed_version {
    ready.insert(...);
    return false;
} else {
    return true;
} 
```